Loop/Managers/DeviceDataManager.swift

```
@@ -992,6 +1000,77 @@ extension DeviceDataManager {
+
+        let exporter = criticalEventLogExportManager.createHistoricalExporter()
+
+        task.expirationHandler = {
+            self.log.default("Invoked critical event log historical export background task expiration handler - cancelling exporter")
+            exporter.cancel()
+        }
+
+        DispatchQueue.global(qos: .background).async {
+            let error = exporter.export()
+            if let error = error {
+                self.log.error("Critical event log historical export errored: %{public}@", String(describing: error))
+            }
+
+            self.scheduleCriticalEventLogHistoricalExportBackgroundTask(isRetry: error != nil)
+            task.setTaskCompleted(success: error == nil)
+
+            self.log.default("Completed critical event log historical export background task")
+        }
+    }
+
+    public func scheduleCriticalEventLogHistoricalExportBackgroundTask(isRetry: Bool = false) {
+        do {
+            let earliestBeginDate = isRetry ? criticalEventLogExportManager.retryExportHistoricalDate() : criticalEventLogExportManager.nextExportHistoricalDate()
+            let request = BGProcessingTaskRequest(identifier: Self.criticalEventLogHistoricalExportBackgroundTaskIdentifer)
+            request.earliestBeginDate = earliestBeginDate
+            request.requiresExternalPower = true
+
+            try BGTaskScheduler.shared.submit(request)
+
+            log.default("Scheduled critical event log historical export background task: %{public}@", ISO8601DateFormatter().string(from: earliestBeginDate))
+        } catch let error {
+            #if IOS_SIMULATOR
+            log.debug("Failed to schedule critical event log export background task due to running on simulator")
+            #else
+            log.error("Failed to schedule critical event log export background task: %{public}@", String(describing: error))
+            #endif
+        }
+    }
+
+    public func removeExportsDirectory() -> Error? {
```

[nit] consider just having this function throw and let users catch the Error (if they really just want to ignore the error, they can always just call it with `try?`).

---

Per earlier discussion, going to leave as-is.

---

✅